contracts/payment/RefundEscrow.sol

```diff
@@ -0,0 +1,91 @@
+pragma solidity ^0.4.23;
+
+import "./ConditionalEscrow.sol";
+import "../ownership/Ownable.sol";
+
+
+/**
+ * @title RefundEscrow
+ * @dev Escrow that holds investor funds for a unique benefitiary, and allows for
+ * either withdrawal by the benefiatiary, or refunds to the investors.
+ */
+contract RefundEscrow is ConditionalEscrow, Ownable {
+  enum State { Active, Refunding, Closed }
+
+  event Closed();
+  event RefundsEnabled();
+  event Refunded(address indexed investor, uint256 weiAmount);
+
+  State public state;
+  address public beneficiary;
+
+  /**
+   * @dev Constructor.
+   * @param _beneficiary The beneficiary of the investments.
+   */
+  constructor(address _beneficiary) public {
+    require(_beneficiary != address(0));
+    beneficiary = _beneficiary;
+    state = State.Active;
+  }
+
+  /**
+   * @dev Stores funds that may later be refunded.
+   * @param _investor The address funds will be sent to if a refund occurs.
+   */
+  function invest(address _investor) payable public {
+    require(state == State.Active);
+    super.deposit(_investor);
+  }
+
+  /**
+   * @dev Disable the base deposit function, use invest instead.
+   */
+  function deposit(address _payee) payable public {
+    revert();
+  }
+
+  /**
+   * @dev Allows for the beneficiary to withdraw their funds, rejecting
+   * further investments.
+   */
+  function close() onlyOwner public {
+    require(state == State.Active);
+    state = State.Closed;
+    emit Closed();
+  }
+
+  /**
+   * @dev Allows for refunds to take place, rejecting further investments.
+   */
+  function enableRefunds() onlyOwner public {
+    require(state == State.Active);
+    state = State.Refunding;
+    emit RefundsEnabled();
+  }
+
+  /**
+   * @dev Withdraws the beneficiary's funds.
+   */
+  function withdraw() public {
```

Comment:
`Escrow#withdraw(address)` has the same name but it does something different. Let's name this one something different. `beneficiaryWithdraw`? I don't know.

Reply:
Agreed. On a related note, `withdraw` and `refund` do the same thing, but `withdraw` emits no events. Should we prevent the user from calling the base function (i.e. override and revert)?

Reply:
Oh, I hadn't noticed that. Hm... I think we should keep `withdraw` instead of adding a new `refund` function. In the case of `RefundableCrowdsale`, the user will use `RefundableCrowdsale#claimRefund` anyway.